saleor/payment/utils.py

```diff
@@ -68,25 +93,22 @@ def create_payment(**payment_data):
     return payment
 
 
-def create_transaction(
-        payment: Payment,
-        token: str,
-        kind: str,
-        is_success: bool,
-        amount: Decimal,
-        currency: str,
-        gateway_response: Optional[Dict] = None) -> Transaction:
-    if not gateway_response:
+def create_transaction(payment: Payment, kind: str, payment_token: str,
+        gateway_response: Dict=None, error_msg=None, **extra_params) -> Transaction:
+    """Creates a Transaction based on transaction kind and gateway response."""
+    if gateway_response is None:
         gateway_response = {}
+
     txn, _ = Transaction.objects.get_or_create(
         payment=payment,
-        token=token,
-        kind=kind,
-        is_success=is_success,
-        amount=amount,
-        currency=currency,
+        kind=gateway_response.get('kind', kind),
+        token=gateway_response.get('transaction_id', payment_token),
```

Comment:
As these fields are required for the gateway response, probably there is no need to provide default values.

Reply:
Defaults are for when the validation fails or an exception happens, and we don't get the response in a format we can process. I assumed we still might want to create a transaction object that something has happened.

Reply:
Makes sense.

But I'm not sure whether these default are proper not not. Probably these defaults are not the actual ones sent to the gateway. For example, I think the amount is actually passed to gateway by ```extra_params```. Maybe we should use these values which are actually used in the gateways as defaults? Othewise we may create transactions with wrong data.

Reply:
That sounds nice. We should try to use same logic.

Reply:
That will be nice! We should try to keep the same logic at the same place.